chalk-solve/src/clauses/builtin_traits.rs

```diff
@@ -16,8 +18,44 @@ pub fn add_builtin_program_clauses<I: Interner>(
     }
 
     match well_known {
-        WellKnownTrait::SizedTrait => { /* TODO */ }
+        WellKnownTrait::SizedTrait => add_sized_program_clauses(db, builder, trait_ref, ty),
         WellKnownTrait::CopyTrait => { /* TODO */ }
         WellKnownTrait::CloneTrait => { /* TODO */ }
     }
 }
+
+fn add_sized_program_clauses<I: Interner>(
+    db: &dyn RustIrDatabase<I>,
+    builder: &mut ClauseBuilder<'_, I>,
+    trait_ref: &TraitRef<I>,
+    ty: &TyData<I>,
+) {
+    let interner = db.interner();
+
+    let (struct_id, substitution) = match ty {
+        TyData::Apply(ApplicationTy {
+            name: TypeName::Struct(struct_id),
+            substitution,
+        }) => (*struct_id, substitution),
+        _ => return,
+    };
+
+    let struct_datum = db.struct_datum(struct_id);
+
+    if struct_datum.binders.map_ref(|b| b.fields.is_empty()).value {
+        builder.push_fact(trait_ref.clone());
+        return;
+    }
+
+    let last_field_ty = struct_datum
```

Comment:
Let's add a comment here -- something like this:

---

To check if a struct type `S<..>` is `Sized`, we only have to look at its last field. This is because the WF checks for structs require that all the other fields must be `Sized`.